docker/server/entrypoint.sh

```diff
@@ -3,164 +3,213 @@
 set -eo pipefail
 shopt -s nullglob
 
-DO_CHOWN=1
-if [ "${CLICKHOUSE_DO_NOT_CHOWN:-0}" = "1" ]; then
-    DO_CHOWN=0
-fi
+# check to see if this file is being run or sourced from another script
+_is_sourced() {
+	# https://unix.stackexchange.com/a/215279
+	[ "${#FUNCNAME[@]}" -ge 2 ] \
+		&& [ "${FUNCNAME[0]}" = '_is_sourced' ] \
+		&& [ "${FUNCNAME[1]}" = 'source' ]
+}
+
+get_variables() {
+    declare -g DO_CHOWN
+    DO_CHOWN=1
+    if [ "${CLICKHOUSE_DO_NOT_CHOWN:-0}" = "1" ]; then
+        DO_CHOWN=0
+    fi
```

Comment:
Move that inside setup_dirs or use CLICKHOUSE_DO_NOT_CHOWN there directly

Reply:
Done